src/inference/src/core.cpp

```diff
@@ -48,11 +48,17 @@ class Core::Impl : public CoreImpl {
 Core::Core(const std::string& xmlConfigFile) {
     _impl = std::make_shared<Impl>();
 
+    auto _registerPlugins = [&]() {
```

Comment:
Why do you need this change?

Reply:
For the moment `Core` ctor may throw next exceptions - ov::Exception, InferenceEngine::Exception (via IE_THROW), std::runtime_error. And previosly, when `register_plugins` was called, IE::Exception was converted to ov::Exception by `OV_CORE_CALL_STATEMENT`. Because I extended number of arguments in 
`RegisterPluginsInRegistry()`, but not for `register_plugins`, I call `RegisterPluginsInRegistry()` directly. So `Core` started throwing IE::Exception, and it led to tests failures

After introducing of `OV_CORE_CALL_STATEMENT` it was like that:
```
OV_CORE_CALL_STATEMENT({
#ifdef OPENVINO_STATIC_LIBRARY
        _impl->RegisterPluginsInRegistry(::getStaticPluginsRegistry());
#else
        // If XML is default, load default plugins by absolute paths
        auto loadByAbsPath = xmlConfigFile.empty();
        _impl->RegisterPluginsInRegistry(findPluginXML(xmlConfigFile), loadByAbsPath);
#endif
})
```
but on Windows it led to build failure:
```
'#': invalid character: possibly the result of a macro expansion
'ifdef': undeclared identifier
syntax error: missing ';' before identifier 'OPENVINO_STATIC_LIBRARY'
'OPENVINO_STATIC_LIBRARY': undeclared identifier
syntax error: missing ';' before identifier '_impl'
'getStaticPluginsRegistry': is not a member of '`global namespace''
'getStaticPluginsRegistry': identifier not found
illegal else without matching if
'endif': undeclared identifier
syntax error: missing ';' before '}'
```

So the final solution is like that

Reply:
I decided to remove lambda and replace it like that:
```
#ifdef OPENVINO_STATIC_LIBRARY
        OV_CORE_CALL_STATEMENT(_impl->RegisterPluginsInRegistry(::getStaticPluginsRegistry());)
#else
        OV_CORE_CALL_STATEMENT({
             // If XML is default, load default plugins by absolute paths
            auto loadByAbsPath = xmlConfigFile.empty();
            _impl->RegisterPluginsInRegistry(findPluginXML(xmlConfigFile), loadByAbsPath);
        })
#endif
```